### DISCUSSION TITLE
Why are all components loaded when refreshing?

### DISCUSSION THREAD

**Discussion by cdwmhcc** [nuxt/nuxt#23573] [Questions]
When using npm run dev to develop locally, all components will be loaded every time the page is refreshed.
I don't remember that this has changed since version 3.x. Is there any option to configure only the modules used by the current page to be loaded?

![image](https://github.com/nuxt/nuxt/assets/8030134/2f6a77b5-1178-4f46-9583-cfa02cb46e8a)
![image](https://github.com/nuxt/nuxt/assets/8030134/7046304d-db31-4979-b521-f356adb6da64)

**Comment by TheAlexLichter** [comment]
Might be related to https://github.com/nuxt/cli/issues/209

**Reply by nazarii-semeniuk** [reply]
I have the same issue, and tried to make steps to fix this from this link, but it doesn't help.

**Comment by nazarii-semeniuk** [comment]
Have the same issue, but my situation is even worse, because I have a lot of components. Every page update takes up to 50s to download all components. Development experience is awful :(
